src/Core/ClassInfo.php

```diff
@@ -560,4 +561,51 @@ public static function parse_class_spec($classSpec)
         static::$_cache_parse[$classSpec] = $result;
         return $result;
     }
+
+
+    /**
+     * Returns a list of classes with a particular extension applied
+     *
+     * This reflects all extensions added (or removed) both via yaml as well as dynamically
+     * using Extensible::add_extension() and Extensible::removed_extension().
+     *
+     * Please not that
+     *
+     * @param string $extensionClass        extension class name
+     * @param string $baseClassOrObject     class or object to find subclasses of with the extension applied
+     * @param bool $includeBaseClass        include the base class itself?
+     * @param bool $includeSubclasses       include all levels of inheritance?
+     * @param bool $excludeTestOnly         exclude classes implementing TestOnly interface?
+     * @return array
+     * @throws \ReflectionException
+     */
+    public static function classesWithExtension(
+        string $extensionClass,
+        string $baseClassOrObject = DataObject::class,
+        bool $includeBaseClass = false,
+        bool $excludeTestOnly = true
+    ) {
+        // get class names
+        $baseClass = self::class_name($baseClassOrObject);
+
+        // get a list of all subclasses for a given class
+        $classes = ClassInfo::subclassesFor($baseClass, $includeBaseClass);
+
+        // include the base class if required
+        if ($includeBaseClass) {
+            $classes = array_merge([strtolower($baseClass) => $baseClass], $classes);
+        }
+
+        // exclude TestOnly if required
+        $classes = array_filter($classes, function ($class) use ($excludeTestOnly) {
+            return !($excludeTestOnly && is_a($class, TestOnly::class, true));
+        });
```

Comment:
This will still loop if `!$excludeTestOnly` but won't actually do anything. You should move the condition here out of the loop.

Reply:
True, it was previously in a loop doing more things and after a tiny refactor stayed there. Moved outside.